Mathlib/CategoryTheory/Shift/Adjunction.lean

```diff
@@ -0,0 +1,461 @@
+
+/--
+Given an adjunction `F ⊣ G` and a `CommShift` structure on `F`, this constructs
+the unique compatible `CommShift` structure on `G`.
+-/
+@[simps]
+noncomputable def rightAdjointCommShift [F.CommShift A] : G.CommShift A where
+  iso a := RightAdjointCommShift.iso adj a
+  zero := by
+    refine CommShift.CompatibilityUnit_unique_right adj (F.commShiftIso 0)  _ _
+      (RightAdjointCommShift.iso_CompatibilityUnit adj 0) ?_
+    rw [F.commShiftIso_zero]
+    exact CommShift.CompatibilityUnit_isoZero adj
+  add a b := by
+    refine CommShift.CompatibilityUnit_unique_right adj (F.commShiftIso (a + b))  _ _
+      (RightAdjointCommShift.iso_CompatibilityUnit adj (a + b)) ?_
+    rw [F.commShiftIso_add]
+    exact CommShift.CompatibilityUnit_isoAdd adj _ _ _ _
+      (RightAdjointCommShift.iso_CompatibilityUnit adj a)
+      (RightAdjointCommShift.iso_CompatibilityUnit adj b)
+
+lemma commShift_of_leftAdjoint [F.CommShift A] :
+    letI := adj.rightAdjointCommShift A
+    adj.CommShift A := by
+  letI := adj.rightAdjointCommShift A
+  apply CommShift.mk'
+  refine ⟨fun a => ?_⟩
+  ext X
+  dsimp
+  simp only [Functor.commShiftIso_id_hom_app, Functor.comp_obj,
+    Functor.id_obj, id_comp, Functor.commShiftIso_comp_hom_app]
+  exact RightAdjointCommShift.iso_CompatibilityUnit adj a X
+
+end Adjunction
+
+namespace Equivalence
+
+variable {C D : Type*} [Category C] [Category D] (E : C ≌ D)
+
+section
+
+variable (A : Type*) [AddMonoid A] [HasShift C A] [HasShift D A]
+
+/--
+If `E : C ≌ D` is an equivalence, this expresses the compatibility of `CommShift`
+structures on `E.functor` and `E.inverse`.
+-/
+abbrev CommShift [E.functor.CommShift A] [E.inverse.CommShift A] : Prop :=
+  E.toAdjunction.CommShift A
+
+namespace CommShift
+
+instance [h : E.functor.CommShift A] : E.symm.inverse.CommShift A := h
+instance [h : E.inverse.CommShift A] : E.symm.functor.CommShift A := h
+
+/-- Constructor for `Equivalence.CommShift`. -/
+lemma mk' [E.functor.CommShift A] [E.inverse.CommShift A]
+    (h : NatTrans.CommShift E.unitIso.hom A) :
+    E.CommShift A where
+  commShift_unit := h
+  commShift_counit := (Adjunction.CommShift.mk' E.toAdjunction A h).commShift_counit
+
+/-- Constructor for `Equivalence.CommShift`. -/
+lemma mk'' [E.functor.CommShift A] [E.inverse.CommShift A]
+    (h : NatTrans.CommShift E.counitIso.hom A) :
+    E.CommShift A where
+  commShift_unit := by
+    have h' := NatTrans.CommShift.of_iso_inv E.counitIso A
+    have : NatTrans.CommShift E.unitIso.symm.hom A :=
+      (Adjunction.CommShift.mk' E.symm.toAdjunction A h').commShift_counit
+    exact NatTrans.CommShift.of_iso_inv E.unitIso.symm A
+  commShift_counit := h
+
+/--
+If `E : C ≌ D` is an equivalence and we have compatible `CommShift` structures on `E.functor`
+and `E.inverse`, then these structures are also compatible with the adjunction
+`E.toAdjunction : E.functor ⊣ E.inverse`.
+-/
+instance [E.functor.CommShift A] [E.inverse.CommShift A] [E.CommShift A] :
```

Comment:
As we an an abbrev, this is no longer necessary.

Reply:
Removed.